prusti-viper/src/encoder/pure_function_encoder.rs

```diff
@@ -929,9 +929,17 @@ impl<'p, 'v: 'p, 'tcx: 'v> BackwardMirInterpreter<'tcx>
                     vir::Expr::not(cond_val)
                 };
 
+                let assert_msg = if let mir::AssertKind::BoundsCheck{ .. } = msg {
+                    // Use the debug impl for BoundsCheck, as it is supposed to be handled before
+                    // calling display() according to the docs
+                    // TODO: use fmt_assert_args once #BUGNR is merged
```

Comment:
will insert the rustc PR number here once i have it.

Reply:
alright, i think we could merge this as is for now. alternatively, we could wait ~1 day and hope the rustc side reacts quickly.

Reply:
Let's merge now